pyaccel/optics.py

```diff
@@ -463,8 +463,17 @@ def get_frac_tunes(accelerator=None, m66=None, closed_orbit=None,
 
 
 @_interactive
-def get_chromaticities(accelerator):
-    raise OpticsException('not implemented')
+def get_chromaticities(accelerator, energy_offset=1e-6):
+    acc = accelerator[:]
+    acc.cavity_on = True
+    acc.radiation_on = True
+    cod = _tracking.find_orbit6(acc)
+    nux, nuy, *_ = get_frac_tunes(acc, closed_orbit=cod)
+    cod[4] += energy_offset
+    nux_den, nuy_den, *_ = get_frac_tunes(acc, closed_orbit=cod)
```

Comment:
I think you have to call `find_orbit6` again, don't you?

Isn't it easier to calculate chromaticity with 4D tracking? Otherwise you'll have to calculate the momentum compaction first, Am I wrong?